### PR TITLE
feat: add nix store gc command

### PR DESCRIPTION
**Pull request by mkenigs**

### CODE REVIEW

**Comment by mkenigs** [comment]
I didn't get the pre-commit checks to pass as `rust-env` appears to be broken on main:
```
> flox build .#devShells.aarch64-darwin.rust-env
error: anonymous function at /nix/store/lndimpmjm6yjvxdlcjqc066z4vy1sl8n-source/pkgs/rustfmt/default.nix:1:1 called with unexpected argument 'asNightly'

       at /nix/store/hc2y6044ksiz4vs1cyhy6qzd6szxvm4c-source/lib/customisation.nix:80:16:

           79|     let
           80|       result = f origArgs;
             |                ^
           81|
(use '--show-trace' to show detailed location information)
ERROR: Error running nix: Nix call unsuccessful: [exit status: 1]
```

File: crates/runix/src/command.rs [hunk 1]
```diff
@@ -227,3 +228,16 @@ impl JsonCommand for Bundle {}
...
+/// `nix store gc` Command
+#[derive(Debug, Default, Clone)]
+pub struct StoreGc {
+    pub store_gc: StoreGcArgs,
+}
+
+impl NixCliCommand for StoreGc {
+    type Own = StoreGcArgs;
+
+    const OWN_ARGS: Group<Self, Self::Own> = Some(|d| d.store_gc.clone());
```
Comment (ysndr) [inline crates/runix/src/command.rs:242]: Let's add a comment that `store gc` does not support any of the other argument Sets, this looks like something is missing..

**Comment by ysndr** [comment]
Ah i should clean that up
You should be able to run `flox develop runix`

**Review by ysndr — APPROVED** [review]
I'll have a look at the devShell/precommit hooks but otherwise this looks fine

**Pull request merged**